k8s.go

```diff
@@ -0,0 +1,250 @@
+package main
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	api "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	defaultK8sNamespace      = "default"
+	defaultK8sResourcePolicy = "keep"
+)
+
+type k8sSecretOptions struct {
+	Namespace       string `long:"namespace" description:"The Kubernetes namespace the secret is located in"`
+	SecretName      string `long:"secret-name" description:"The name of the Kubernetes secret"`
+	SecretEntryName string `long:"secret-entry-name" description:"The name of the entry within the secret"`
```

Comment:
Same concern regarding the naming here. I realize "entry" is used throughout, so I won't leave additional comments about it. Seems like a significant change, so we can always reconsider it at the next version bump instead.